tardis/tests/tests_slow/test_w7.py

```diff
@@ -39,11 +37,66 @@ class TestW7(object):
         * last_line_interaction_shell_id |     * montecarlo_nu
         * nubar_estimators               |     * last_line_interaction_angstrom
         * ws                             |     * j_blues_norm_factor
+
+    * spectrum.npz
+    Contents (all (.npy)):
+        * luminosity_density_nu          | * wavelength
+        * delta_frequency                | * luminosity_density_lambda
+    """
+
+    # TODO: make this fixture ingest data from an HDF5 file.
+    slow_test_data_dir = os.path.join(os.path.expanduser(
+            os.path.expandvars(request.config.getvalue('slow-test-data'))),
+            "w7")
+
+    ndarrays = dict(np.load(os.path.join(slow_test_data_dir, "ndarrays.npz")))
+    quantities = dict(np.load(os.path.join(slow_test_data_dir, "quantities.npz")))
+    spectrum = dict(np.load(os.path.join(slow_test_data_dir, "spectrum.npz")))
+
+    # Associate CGS units to ndarrays of baseline quantities.
+    quantities['j_blues_norm_factor'] = \
+        quantities['j_blues_norm_factor'] * u.Unit('1 / (cm2 s)')
+
+    quantities['last_line_interaction_angstrom'] = \
+        quantities['last_line_interaction_angstrom'] * u.Unit('Angstrom')
+
+    quantities['luminosity_inner'] = \
+        quantities['luminosity_inner'] * u.Unit('erg / s')
+
+    quantities['montecarlo_luminosity'] = \
+        quantities['montecarlo_luminosity'] * u.Unit('erg / s')
+
+    quantities['montecarlo_virtual_luminosity'] = \
+        quantities['montecarlo_virtual_luminosity'] * u.Unit('erg / s')
+
+    quantities['montecarlo_nu'] = quantities['montecarlo_nu'] * u.Unit('Hz')
+    quantities['t_rads'] = quantities['t_rads'] * u.Unit('K')
+
+    # Associate CGS units to ndarrays of baseline quantities (spectrum).
+    spectrum['luminosity_density_nu'] = spectrum['luminosity_density_nu'] * u.Unit('erg')
+    spectrum['delta_frequency'] = spectrum['delta_frequency'] * u.Unit('Hz')
+    spectrum['wavelength'] = spectrum['wavelength'] * u.Unit('Angstrom')
+
+    spectrum['luminosity_density_lambda'] = \
+            spectrum['luminosity_density_lambda'] * u.Unit('erg / (Angstrom s)')
+
+    return dict(ndarrays.items() + quantities.items() + spectrum.items())
```

Comment:
I'm not sure this a good way to combine dictionaries. I think using `dict.update` would be the safer approach.